src/main/java/se/sundsvall/document/integration/db/model/DocumentTypeEntity.java

```diff
@@ -0,0 +1,175 @@
+package se.sundsvall.document.integration.db.model;
+
+import static org.hibernate.annotations.TimeZoneStorageType.NORMALIZE;
+
+import java.io.Serializable;
+import java.time.OffsetDateTime;
+import java.util.Objects;
+
+import org.hibernate.annotations.TimeZoneStorage;
+
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.EntityListeners;
+import jakarta.persistence.Id;
+import jakarta.persistence.IdClass;
+import jakarta.persistence.Index;
+import jakarta.persistence.Table;
+import se.sundsvall.document.integration.db.model.listener.DocumentTypeEntityListener;
+
+@Entity
+@IdClass(DocumentTypeId.class)
+@Table(name = "document_type", indexes = {
+	@Index(name = "ix_municipality_id_type", columnList = "municipality_id, `type`"),
+	@Index(name = "ix_municipality_id", columnList = "municipality_id")
+})
+@EntityListeners(DocumentTypeEntityListener.class)
+public class DocumentTypeEntity implements Serializable {
+
+	private static final long serialVersionUID = -4452832623957756766L;
+
+	@Id
+	@Column(name = "municipality_id")
+	private String municipalityId;
+
+	@Id
+	@Column(name = "`type`", nullable = false)
+	private String type;
```

Comment:
I don't think this is the way to go. If the type is ever changeable this might lead to some issues and I could definitely see a case where the clients would like to rename a type.

You could probably get the desired result by having a auto-generated primary key and using unique constraints for type + municipality.

Reply:
Adjusted to use calculated id column instead